pyproject.toml

```diff
@@ -1,7 +1,70 @@
 [build-system]
-requires = ["setuptools>=40.8.0", "wheel", "Cython", "py-cpuinfo"]
-build-backend = "setuptools.build_meta:__legacy__"
+requires = [
+    "setuptools>=64",
+    "setuptools_scm[toml]>=6.2",
+    "wheel",
+    "Cython",
+    "py-cpuinfo"
+]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "numcodecs"
+description = """
+A Python package providing buffer compression and transformation codecs for use
+in data storage and communication applications.
+"""
+readme =  "README.rst"
+dependencies = [
+    "entrypoints",
+    "numpy>=1.7",
+    "typing-extensions>=3.7.4"
+]
+requires-python = ">=3.7, <4"
```

Comment:
This was already in `setup.py`

https://github.com/zarr-developers/numcodecs/blob/main/setup.py#L321

That said, no strong feelings here

Reply:
True. But removing it would be more readable and consistent with  zarr-python:
https://github.com/zarr-developers/zarr-python/blob/bbc66df11ef0c731ebece330c1e60c38f1f36aed/pyproject.toml#L13

But no strong feelings either. And it could be deferred to another PR any way.

Reply:
Let's give others some time to comment. If there are no objections, will drop it.

Reply:
Went ahead and dropped it. Though will leave this thread open in case others have thoughts